assets/src/components/map.tsx

```diff
@@ -598,7 +601,11 @@ const Map = (props: Props): ReactElement<HTMLDivElement> => {
           <TrainVehicle key={trainVehicle.id} trainVehicle={trainVehicle} />
         ))}
         {(props.shapes || []).map((shape) => (
-          <Shape key={shape.id} shape={shape} />
+          <Shape
+            key={shape.id}
+            shape={shape}
+            direction={props.stopCardDirection}
```

Comment:
I like the idea of adding it to the shape, I'll give that a look.

Reply:
Posting as much to work my own thoughts out as to respond:

It looks like the main difficulty is going to be that, while in our own data we have separate shapes for separate directions, there's actually nothing in the GTFS spec that inherently links a shape to a single direction - the same shape could be associated with trips in both directions. So when we're doing `useTripShape` we could easily fill in the direction of the current trip, but for `useRouteShapes` it's a bit less obvious, unless we add additional GTFS parsing code to associate directions with shapes and then handle any edge cases in case they ever come up. We could also only add that field when returning the shape from `useTripShape`, but that means the field would have to be nullable, and I don't like introducing things where different ways of fetching the same object yield subtly different results.

Reply:
Thanks for considering it! Fully agree with
> I don't like introducing things where different ways of fetching the same object yield subtly different results.

I do wonder if it would be worth keeping `useTripShape` and all the BE data the same, and the consumer of `useTripShape` could transform the `Shape` into a `ShapeWithDirection` as only a type on the FE. Though I'm not sure that is much different from "different ways of fetching the same object yield subtly different results".

Also working out my own thoughts on this, so that is definitely a non-blocking suggestion!

Reply:
It's a shame that this isn't more straightforward, it would make some of this code so much cleaner!

Reply:
@KaylaBrady okay, I created [a followup ticket](https://app.asana.com/0/1148853526253420/1203626333738235/f) to make the changes we've been discussing. Let me know what you think!

Reply:
Looks great, thanks!!